outlines/serve/serve.py

```diff
@@ -122,6 +127,8 @@ async def stream_results() -> AsyncGenerator[bytes, None]:
     # Sets default for the model (`facebook/opt-125m`)
     engine = AsyncLLMEngine.from_engine_args(engine_args)
 
+    _adapt_tokenizer(engine.engine.tokenizer)
```

Comment:
Why are you calling this function here? The result is not used.

Reply:
The tokenizer is changed inside the function anyways. I now assigned it to the tokenizer though.

Reply:
Ah makes sense. It's not needed however as vLLM handles tokenisation on its end during encoding/decoding.

Reply:
It's needed because in here https://github.com/outlines-dev/outlines/blob/fde61a80a58de0401fdecdee7408db53e17ca4f4/outlines/fsm/fsm.py#L345 outlines expects tokenizer to return a list but vllm tokenizers return string. 
Also I just realized that this change makes a breaking change to the library. If it's fine by the project directors its fine, if not we might need to revert and re-think about this.